BentoKit/BentoKit/Screen/ScreenLifecycleEvent.swift

```diff
@@ -0,0 +1,38 @@
+public protocol ScreenLifecycleObserving {
+    func send(_ event: ScreenLifecycleEvent)
```

Comment:
Do you have any thoughts about merging this protocol with `BoxViewModel` and its `func send(action: Action)` ?

It will be a bigger breaking change, but IMO having 2 separate protocols with 2 separate `func send` interfaces are redundant.

If we end up keeping this protocol, please add documentation for `ScreenLifecycleObserving` that requires ViewModel to be conformed alongside `BoxViewModel`.

Reply:
I don't mind.

Reply:
^ Above will look like this:

```swift
public protocol BoxViewModel: AnyObject {
    associatedtype State
    associatedtype Action

    var state: Property<State>  { get }

    func send(action: ViewLifeCycleOrUserAction<Action>)
}

enum ViewLifeCycleOrUserAction<Action> {
    case viewLifecycle(ScreenLifecycleEvent)
    case userAction(Action)
}
```

Reply:
I prefer to keep them separate, like, they are from separated sources after all. Little point to glue them together artificially.